rskj-core/src/main/java/org/ethereum/datasource/RocksDbKeyIterator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.datasource;
+
+import org.ethereum.db.ByteArrayWrapper;
+import org.ethereum.util.ByteUtil;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksIterator;
+
+import java.util.NoSuchElementException;
+
+public class RocksDbKeyIterator implements DataSourceKeyIterator {
+    private RocksIterator iterator;
+
+    public RocksDbKeyIterator(RocksDB db) {
+        this.iterator = db.newIterator();
```

Comment:
It would be needed to position the iterator at the beginning before using it calling seekToFirst()

From the api
> public [RocksIterator](https://javadoc.io/static/org.rocksdb/rocksdbjni/6.27.3/org/rocksdb/RocksIterator.html) newIterator()
Return a heap-allocated iterator over the contents of the database. The result of newIterator() is initially invalid (caller must call one of the Seek methods on the iterator before using it).

Caller should close the iterator when it is no longer needed. The returned iterator should be closed before this db is closed.

Reply:
Yeah looks like I forgot to put it there. Done